File: packages/js/core/src/uri-resolution/resolvers/redirects/RedirectsResolver.ts
import { applyRedirects } from "../../../algorithms";
import { Uri, Client } from "../../../types";
import { UriResolver, ResolveUriResult } from "../../core";

export class RedirectsResolver implements UriResolver<void> {
  public get name(): string {
    return RedirectsResolver.name;
  }

  async tryResolveToWrapper(
    uri: Uri,
    client: Client
  ): Promise<ResolveUriResult<void>> {
    const redirectedUri = applyRedirects(uri, client.getRedirects({}));

    return Promise.resolve({
      uri: redirectedUri,
    });
  }
}

File: packages/py/polywrap-core/polywrap_core/uri_resolution/resolvers/redirects/redirects_resolver.py
from __future__ import annotations

from ....algorithms import apply_redirects
from ....types import Client, Uri, WrapperCache
from ...core import UriResolutionResult, UriResolutionStack, UriResolver


class RedirectsResolver(UriResolver):
    @property
    def name(self) -> str:
        return type(self).__name__

    async def resolve_uri(
        self,
        uri: Uri,
        client: Client,
        cache: WrapperCache,
        resolution_path: UriResolutionStack,
    ) -> UriResolutionResult:
        redirect_uri = apply_redirects(
            uri, client.get_redirects()
        )

        return UriResolutionResult(uri=redirect_uri)
